coral-spark/src/main/java/com/linkedin/coral/spark/transformers/ExtractUnionFunctionTransformer.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.transformers;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNumericLiteral;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import com.linkedin.coral.com.google.common.collect.ImmutableList;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.hive.hive2rel.functions.CoalesceStructUtility;
+import com.linkedin.coral.spark.containers.SparkUDFInfo;
+
+
+/**
+ * This transformer transforms `extract_union` to `coalesce_struct`.
+ * Instead of leaving `extract_union` visible to Spark, since we adopted the new exploded struct schema (a.k.a struct_tr)
+ * that is different from extract_union's output (a.k.a struct_ex) to interpret union in Coral IR,
+ * we need to swap the reference of `extract_union` to a new UDF that is coalescing the difference between
+ * struct_tr and struct_ex.
+ * See {@link CoalesceStructUtility#COALESCE_STRUCT_FUNCTION_RETURN_STRATEGY} and its comments for more details.
```

Comment:
the documentation is hard to follow, plus the documentation on `CoalesceStructUtility` suggests `coalesce the {@link RelDataType} of struct between Trino's representation and hive's extract_union UDF's representation`. No mention of hive -> spark transformation. 

Implementation of this transformer looks good. Could you take another stab at documentation? maybe add an example like `CastToNamedStructTransformer`

Reply:
Thanks for suggestion, added `Spark` to `CoalesceStructUtility` java doc.
As for the example, it mentioned that `Check `CoralSparkTest#testUnionExtractUDF` for examples.`